src/EditorFeatures/CSharpTest/Interactive/NavigateTo/InteractiveNavigateToTests.cs

```diff
@@ -97,11 +102,12 @@ class Goo<T> where T : IEnumerable
             {
                 var item = (await _aggregator.GetItemsAsync("Goo")).Single(x => x.Kind != "Method");
                 VerifyNavigateToResultItem(item, "Goo", "[|Goo|]<T>", PatternMatchKind.Exact, NavigateToItemKind.Class, Glyph.ClassPrivate);
-            });
+            }, trackingServiceType);
```

Comment:
it seems very unfortunate that every test has to effectively be parameterized itself on how it permute the workspace mef container.  But if this really is the most effective way to accomplish this, then i suppose that's that.

Reply:
I'll look into the possibility of parameterizing the *constructor* instead, but for now I believe this is the best option we have available and implemented.

Reply:
If you could spend a few minutes trying to parameterize the constructor... that would be great.  It would make this a lot more palatable for me if that coudl work out.  If it doesn't... c'est la vie.  But i think that would help with one of my personal goals (though not the primary goal) of keeping tests concise, with much less ceremony.

Reply:
➡️ I'm 95% confident it's not a task to be considered at this scale. It's a future work kind of thing to assist in some of the overall goals.

With that said, there is another way to do this without parameterizing the tests: make the `Factory` type a mutable service, and reassign a `Func<Workspace, IDocumentTrackingService>` in the factory between each test case.

Reply:
i thnk i might prefer the latter.  because then all the tests can stay simple and just call one TestASync function.  Then all the complexity and configuration can happen in the TestAsync function, and will not appear in 1000 top level tests themselves.

Reply:
@CyrusNajmabadi I'll add a commit to the top of this pull request that switches to that strategy. If we like it more we'll keep it, otherwise it'll be easy to drop back out.